master/internal/api_experiment.go

```diff
@@ -2093,23 +2093,18 @@ func sortExperiments(sortString *string, experimentQuery *bun.SelectQuery) error
 			experimentQuery.OrderExpr(
 				fmt.Sprintf("e.validation_metrics->'%s' %s",
 					metricName, sortDirection))
-		case strings.HasPrefix(paramDetail[0], "training."):
-			metricDetails := strings.Split(paramDetail[0], ".")
-			metricQualifier := metricDetails[len(metricDetails)-1]
-			metricName := strings.TrimSuffix(
-				strings.TrimPrefix(paramDetail[0], "training."),
-				"."+metricQualifier)
-			if !slices.Contains(SummaryMetricStatistics, metricQualifier) {
-				return status.Errorf(codes.InvalidArgument,
-					"sort training metrics by statistic: last, max, min, or mean")
+		case strings.Contains(paramDetail[0], "."):
+			metricGroup, metricName, metricQualifier, err := parseMetricsName(paramDetail[0])
+			if err != nil {
+				return err
 			}
 			if metricQualifier == "mean" { //nolint: goconst
-				locator := bun.Safe("trials.summary_metrics->'avg_metrics'")
+				locator := bun.Safe(fmt.Sprintf("trials.summary_metrics->'%s'", metricGroup))
```

Comment:
Not sure if this is okay

Reply:
metricGroup is user input, so it shouldn't be marked as safe with `bun.Safe`. instead, add it as an argument to the next line.